### PR TITLE
feat: SDA-1995: implement download manager for Mana

### PR DESCRIPTION
**Pull request by VishwasShashidhar**
## Description
Implement Download Manager for Mana
[SDA-1995](https://perzoinc.atlassian.net/browse/SDA-1995)

## Solution Approach
- Add new methods in the app bridge

## Related PRs
N/A

### CODE REVIEW

File: src/app/download-handler.ts [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    }
+
+    /**
+     * Send download completed event to the renderer process
+     */
+    private sendDownloadCompleted(): void {
+        if (this.window && !this.window.isDestroyed()) {
+            logger.info(`download-handler: Download completed! Informing the client!`);
+            this.window.send('download-completed', this.items.map((item) => {
+                return {id: item._id, fileDisplayName: item.fileDisplayName};
```
Comment (KiranNiranjan) [inline src/app/download-handler.ts:138]: I think it would be better if we include the filesize

File: src/demo/index.html [hunk 2]
```diff
@@ -211,6 +214,11 @@ <h1>Symphony Electron API Demo</h1>
                 sendLogs: 'send-logs',
                 registerAnalyticHandler: 'register-analytic-handler',
                 registerActivityDetection: 'register-activity-detection',
+                registerDownloadHandler: 'register-download-handler',
+                openDownloadItem: 'open-download-item',
+                showDownloadItem: 'show-download-item',
+                clearDownloadItems: 'clear-download-items',
+                clearDownloadItems: 'clear-download-items',
```
Comment (KiranNiranjan) [inline src/demo/index.html:221]: duplicate

File: src/renderer/app-bridge.ts [hunk 3]
```diff
@@ -61,12 +61,13 @@ export class AppBridge {
...
         onAnalyticsEventCallback: (data) => this.analyticsEventCallback(data),
         restartFloater: (data) => this.restartFloater(data),
+        onDownloadItemCallback: (data) => this.onDownloadItemCallback(data),
     };
 
     constructor() {
         // starts with corporate pod and
         // will be updated with the global config url
-        const currentWindow = remote.getCurrentWindow();
+        // const currentWindow = remote.getCurrentWindow();
```
Comment (KiranNiranjan) [inline src/renderer/app-bridge.ts:70]: Should this be commented?

File: src/app/download-handler.ts (hunk 1 above)
Comment (VishwasShashidhar) [reply src/app/download-handler.ts:138]: Do we show file size today in the download manager?

Comment (KiranNiranjan) [reply src/app/download-handler.ts:138]: Yes

**Pull request merged**